src/Std/Data/DHashMap/Raw.lean

```diff
@@ -451,6 +451,13 @@ only those mappings where the function returns `some` value.
     Raw₀.map f ⟨m, h⟩
   else ∅ -- will never happen for well-formed inputs
 
+@[inline] def mapKeyValueInPlace
```

Comment:
This function is definitely useful and I am happy to have it, but I don't think it will help you define `attach`. Probably you need a variant where `f` receives the necessary membership proofs, similar to how we have both `List.map` and `List.pmap`.

Reply:
Do you think it's useful to define `mapEntries` in the case where the hash is preserved, or only `pmap`?